[Reland][Inductor] Prune configs that require more shared memory than the hardware limit

wychi opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #161816

Summary:
This is a re-land of [PR161040](https://github.com/pytorch/pytorch/pull/161040), which had previously caused test failures on AMD GPUs. The tests are now configured to target only NVIDIA GPUs.

This diff removes configurations that exceed the hardware shared memory limit, which causes the following compilation error:
```
No valid triton configs. OutOfMemoryError: out of resource: triton_mm Required: 327680 Hardware limit:232448 Reducing block sizes or `num_stages` may help.
```

Test Plan:

```
pytest test/inductor/test_max_autotune.py
pytest test/inductor/test_triton_heuristics.py

```

Reviewers:

Subscribers:

Tasks:

Tags:

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben

Differential Revision: [D81400958](https://our.internmc.facebook.com/intern/diff/D81400958)

wychi committed:
[Reland][Inductor] Prune configs that require more shared memory than the hardware limit

Summary:
This is a re-land of [PR161040](https://github.com/pytorch/pytorch/pull/161040), which had previously caused test failures on AMD GPUs. The tests are now configured to target only NVIDIA GPUs.

This diff removes configurations that exceed the hardware shared memory limit, which causes the following compilation error:
```
No valid triton configs. OutOfMemoryError: out of resource: triton_mm Required: 327680 Hardware limit:232448 Reducing block sizes or `num_stages` may help.
```

Test Plan:

```
pytest test/inductor/test_max_autotune.py
pytest test/inductor/test_triton_heuristics.py

```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

coconutruben commented on torch/_inductor/template_heuristics/triton.py:
```diff
@@ -541,34 +541,68 @@ def _scale_mm_configs(
+        If the device does not report available shared memory, returns None.
+        """
+
+        try:
```
nit: I think you can simplify this by just trusting the try/except

coconutruben commented on torch/_inductor/template_heuristics/triton.py:
```diff
@@ -541,34 +541,68 @@ def _scale_mm_configs(
+            # If CUDA is not available or properties cannot be queried, return None
+            return None
+
+        def exceeds(gemm_config: BaseConfig, dtype_size: int) -> bool:
```
nit: can you add a TODO here to make a BaseDeviceConfigHeuristics and have CUDA as an implementation of that? this is fine for the PR now, but I don't like how many if/else we have on cuda vs rocm vs whatever, this should happen at the class level

coconutruben commented:
code is fine to me, let's make sure all the devices pass the tests

wychi committed:
Update on "[Reland][Inductor] Prune configs that require more shared memory than the hardware limit"

Summary:
This is a re-land of [PR161040](https://github.com/pytorch/pytorch/pull/161040), which had previously caused test failures on AMD GPUs. The tests are now configured to target only NVIDIA GPUs.

This diff removes configurations that exceed the hardware shared memory limit, which causes the following compilation error:
```
No valid triton configs. OutOfMemoryError: out of resource: triton_mm Required: 327680 Hardware limit:232448 Reducing block sizes or `num_stages` may help.
```

Test Plan:

```
pytest test/inductor/test_max_autotune.py
pytest test/inductor/test_triton_heuristics.py

```

Reviewers:

Subscribers:

Tasks:

Tags:

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

wychi committed:
Update on "[Reland][Inductor] Prune configs that require more shared memory than the hardware limit"


Summary:
This is a re-land of [PR161040](https://github.com/pytorch/pytorch/pull/161040), which had previously caused test failures on AMD GPUs. The tests are now configured to target only NVIDIA GPUs.

This diff removes configurations that exceed the hardware shared memory limit, which causes the following compilation error:
```
No valid triton configs. OutOfMemoryError: out of resource: triton_mm Required: 327680 Hardware limit:232448 Reducing block sizes or `num_stages` may help.
```

Test Plan:

```
pytest test/inductor/test_max_autotune.py
pytest test/inductor/test_triton_heuristics.py

```

Reviewers:

Subscribers:

Tasks:

Tags:

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

wychi committed:
Update on "[Reland][Inductor] Prune configs that require more shared memory than the hardware limit"


Summary:
This is a re-land of [PR161040](https://github.com/pytorch/pytorch/pull/161040), which had previously caused test failures on AMD GPUs. The tests are now configured to target only NVIDIA GPUs.

This diff removes configurations that exceed the hardware shared memory limit, which causes the following compilation error:
```
No valid triton configs. OutOfMemoryError: out of resource: triton_mm Required: 327680 Hardware limit:232448 Reducing block sizes or `num_stages` may help.
```

Test Plan:

```
pytest test/inductor/test_max_autotune.py
pytest test/inductor/test_triton_heuristics.py

```

Reviewers:

Subscribers:

Tasks:

Tags:

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

wychi commented:
@pytorchbot merge this

wychi commented:
@pytorchbot merge

wychi commented:
@wychi has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D81400958).

wychi commented:
close this PR as it seems cause some problems to CI tools. Will restart a new PR from pure OSS workflow.
